packages/engine/Source/Renderer/TextureAtlas.js

```diff
@@ -0,0 +1,802 @@
+    colorTextures: [oldTexture],
+    destroyAttachments: false,
+  });
+
+  gl.activeTexture(gl.TEXTURE0);
+  gl.bindTexture(target, newTexture._texture);
+
+  framebuffer._bind();
+
+  // Copy any textures from the old atlas to its new position in the new atlas
+  const oldRectangles = this.rectangles;
+  const subRegions = this._subRegions;
+  for (let index = 0; index < oldRectangles.length; ++index) {
+    const rectangle = rectangles[index];
+    const frameBufferOffset = oldRectangles[index];
+
+    if (
+      !defined(rectangle) ||
+      !defined(frameBufferOffset) ||
+      defined(subRegions.get(index)) // The rectangle corresponds to a subregion of a parent image
+    ) {
+      continue;
+    }
+
+    const { x, y, width, height } = rectangle;
+    gl.copyTexSubImage2D(
+      target,
+      0,
+      x,
+      y,
+      frameBufferOffset.x,
+      frameBufferOffset.y,
+      width,
+      height,
+    );
+  }
+
+  gl.bindTexture(target, null);
+  newTexture._initialized = true;
+
+  framebuffer._unBind();
+  framebuffer.destroy();
+
+  return newTexture;
+};
+
+/**
+ * Recreates the texture atlas texture with new dimensions and repacks images as needed.
+ * @param {Context} context The rendering context
+ * @param {number} [queueOffset = 0] Index of the last queued item that was successfully packed
+ * @private
+ */
+TextureAtlas.prototype._resize = function (context, queueOffset = 0) {
+  const borderPadding = this._borderWidthInPixels;
+  const oldRectangles = this._rectangles;
+  const queue = this._imagesToAddQueue;
+
+  const oldTexture = this._texture;
+  let width = oldTexture.width;
+  let height = oldTexture.height;
+
+  // Get the rectangles (width and height) of the current set of images,
+  // ignoring the subregions, which don't get packed
+  const subRegions = this._subRegions;
+  const toPack = oldRectangles
+    .map((image, index) => {
+      return new AddImageRequest({ index, image });
+    })
+    .filter(
+      (request, index) =>
+        defined(request.image) && !defined(subRegions.get(index)),
+    );
+
+  // Add the new set of images
+  let maxWidth = 0;
+  let maxHeight = 0;
+  let areaQueued = 0;
+  for (let i = queueOffset; i < queue.length; ++i) {
+    const { width, height } = queue[i].image;
+    maxWidth = Math.max(maxWidth, width);
+    maxHeight = Math.max(maxHeight, height);
+    areaQueued += width * height;
+    toPack.push(queue[i]);
+  }
+
+  // At minimum, the texture will need to scale to accomodate the largest width and height
+  width = Math.max(maxWidth, width);
+  height = Math.max(maxHeight, height);
+
+  if (!context.webgl2) {
+    width = CesiumMath.nextPowerOfTwo(width);
+    height = CesiumMath.nextPowerOfTwo(height);
+  }
+
+  // Determine by what factor the texture need to be sclaed by at minimum
+  const areaDifference = areaQueued;
+  let scalingFactor = 1.0;
+  while (areaDifference / width / height >= 1.0) {
+    scalingFactor *= 2.0;
+
+    // Resize by one dimension
```

Comment:
is resizing the atlas by one dimension covered in the unit tests? (I think the answer is yes)

Reply:
Yep!